File: test/data/cs/test04/StateIdle.cs
public class StateIdle : BaseState
{
    private StateController _stateController; 
    private IControllee _controllee; 
    public StateIdle(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitEvConfig()
    {
        _controllee.DoAction1();
        _stateController.InstanceOfConfiguring.Setup();
        return _stateController.InstanceOfConfiguring; 
    }
    public override string GetStateName()
    {
        return "NotShooting.Idle"; 
    }
}

File: test/data/typescript/test04/StateIdle.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateIdle extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitEvConfig(): BaseState | null
    {
        this._controlleeImp.DoAction1();
        this._stateController.InstanceOfConfiguring.Setup(false, false);
        return this._stateController.InstanceOfConfiguring;
    }
    public  GetStateName(): string
    {
        return "NotShooting.Idle";
    }
    public GetParent(): BaseState | null
    {
        return this._stateController.InstanceOfNotShooting;
    }
}